### PR TITLE
[CCCP-385] registration-pool migration process

### PR DESCRIPTION
**Pull request by alstjd0921**
## Description

* `KeypairStorage` restructured
* reload `KeypairStorage` on migration started with next pool round
* stop sign psbt | sign only for migration psbt
* only submit next round's system vault pubkey on migration sequence

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Something else (simple changes that are not related to existing functionality or others)

# Checklist

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have made new test codes regarding to my changes.
- [ ] I have no personal secrets or credentials described on my changes.
- [ ] I have run `cargo-clippy`  and linted my code.
- [ ] My changes generate no new warnings.
- [ ] My changes passed the existing test codes.
- [ ] My changes are able to compile.

### CODE REVIEW

File: client/src/btc/handlers/psbt_signer.rs [hunk 1]
```diff
@@ -4,11 +4,15 @@ use br_primitives::{
...
-use miniscript::bitcoin::Psbt;
+use miniscript::bitcoin::{Address as BtcAddress, Network, Psbt};
+use std::str::FromStr;
 use tokio::sync::broadcast::Receiver;
 use tokio_stream::StreamExt;
 
+use br_primitives::substrate::{CustomConfig, MigrationSequence};
 use std::sync::Arc;
+use subxt::OnlineClient;
+use tokio::sync::RwLock;
```
Comment (dnjscksdn98) [inline client/src/btc/handlers/psbt_signer.rs:15]: import 중첩 가능한 것들 묶어주세요

File: client/src/btc/handlers/psbt_signer.rs [hunk 2]
```diff
@@ -115,6 +147,40 @@ impl<T: JsonRpcClient> PsbtSigner<T> {
...
+			.contract_call(
+				registration_pool.vault_address(registration_pool.address(), round),
+				"registration_pool.vault_address",
+			)
+			.await;
+
+		BtcAddress::from_str(&system_vault)
+			.unwrap()
+			.require_network(self.btc_network)
+			.unwrap()
```
Comment (dnjscksdn98) [inline client/src/btc/handlers/psbt_signer.rs:165]: 상관은 없지만 풀에 등록된 모든 주소들은 네트워크 체크가 이미 된거라 여기서는 assume_check()만 써도 되긴 합니다

File: client/src/btc/handlers/psbt_signer.rs [hunk 3]
```diff
@@ -73,10 +88,27 @@ impl<T: JsonRpcClient> PsbtSigner<T> {
...
+				return None;
+			},
+			MigrationSequence::UTXOTransfer => {
+				// Ensure that the unsigned transaction is for the system vault.
+				let system_vault = self.get_system_vault(self.get_current_round().await + 1).await;
+				if !unsigned_psbt.unsigned_tx.output.iter().all(|x| {
+					BtcAddress::from_script(x.script_pubkey.as_script(), self.btc_network).unwrap()
+						== system_vault
+				}) {
+					return None;
```
Comment (dnjscksdn98) [inline client/src/btc/handlers/psbt_signer.rs:107]: 에러 로그랑 센트리 메시지 정도 남겨놔도 괜찮을것 같아요

File: client/src/btc/handlers/psbt_signer.rs [hunk 4]
```diff
@@ -115,6 +147,40 @@ impl<T: JsonRpcClient> PsbtSigner<T> {
...
+		self.get_sub_client()
+			.storage()
+			.at_latest()
+			.await
+			.unwrap()
+			.fetch(&bifrost_runtime::storage().btc_registration_pool().current_round())
+			.await
+			.unwrap()
+			.unwrap()
+	}
```
Comment (dnjscksdn98) [inline client/src/btc/handlers/psbt_signer.rs:183]: sub_client를 여기서만 접근하는 것 같은데, 없애고 client로 대체하고 precompile 접근으로 가져오는건 안되나요?

File: periodic/src/keypair_storage_manager.rs [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+use br_client::eth::EthClient;
+use br_primitives::constants::schedule::MIGRATION_DETECTOR_SCHEDULE;
+use cron::Schedule;
+use ethers::prelude::JsonRpcClient;
+use std::sync::Arc;
+use subxt::OnlineClient;
+use tokio::sync::RwLock;
+use br_client::btc::storage::keypair::KeypairStorage;
+use br_primitives::substrate::bifrost_runtime::btc_registration_pool::storage::types::service_state::ServiceState;
```
Comment (dnjscksdn98) [inline periodic/src/keypair_storage_manager.rs:13]: import 정리

File: periodic/src/keypair_storage_manager.rs [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+			.as_ref()
+			.unwrap()
+			.storage()
+			.at_latest()
+			.await
+			.unwrap()
+			.fetch(&bifrost_runtime::storage().btc_registration_pool().current_round())
+			.await
+			.unwrap()
+			.unwrap()
```
Comment (dnjscksdn98) [inline periodic/src/keypair_storage_manager.rs:70]: 이건 별다른 에러 핸들링은 필요 없을까요?

File: periodic/src/keypair_storage_manager.rs [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+use br_primitives::constants::schedule::MIGRATION_DETECTOR_SCHEDULE;
+use cron::Schedule;
+use ethers::prelude::JsonRpcClient;
+use std::sync::Arc;
+use subxt::OnlineClient;
+use tokio::sync::RwLock;
+use br_client::btc::storage::keypair::KeypairStorage;
+use br_primitives::substrate::bifrost_runtime::btc_registration_pool::storage::types::service_state::ServiceState;
+
+pub struct KeypairStorageManager<T> {
```
Comment (dnjscksdn98) [inline periodic/src/keypair_storage_manager.rs:15]: KeypairMigrator?

**Review by dnjscksdn98 — COMMENTED** [review]
혹시 UTXOTransfer에 필요한 psbt는 pallet에 제출을 어떻게 하나요?
이것도 릴레이어가 socket_queue::unsigned_psbts() 함수로 가져오긴 해서요

**Review by SunoukMun — APPROVED** [review]

**Review by dnjscksdn98 — APPROVED** [review]

**Pull request merged**